misk-admin/src/main/kotlin/misk/web/dashboard/MiskWebTabIndexAction.kt

```diff
@@ -0,0 +1,92 @@
+package misk.web.dashboard
+
+import kotlinx.html.body
+import kotlinx.html.div
+import kotlinx.html.head
+import kotlinx.html.html
+import kotlinx.html.id
+import kotlinx.html.link
+import kotlinx.html.script
+import kotlinx.html.style
+import misk.exceptions.NotFoundException
+import misk.hotwire.buildHtml
+import misk.web.Get
+import misk.web.PathParam
+import misk.web.ResponseContentType
+import misk.web.actions.WebAction
+import misk.web.mediatype.MediaTypes
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Kotlin backed tab loader, equivalent to /_tab/slug/index.html
+ */
+@Singleton
+class MiskWebTabIndexAction @Inject constructor(
+  @AdminDashboard private val dashboardTabs: List<DashboardTab>,
+) : WebAction {
+  @Get("$PATH/{slug}")
+  @ResponseContentType(MediaTypes.TEXT_HTML)
+  @AdminDashboardAccess
+  fun get(@PathParam slug: String?): String {
+    val dashboardTab = dashboardTabs.firstOrNull { slug == it.slug }
```

Comment:
nit: would be better to throw the 404 here, rather than mid way through building the HTML.

```suggestion
    // TODO remove this hack when new Web Actions tab lands and old ones removed, v1 and v2 are in the same web-actions tab
    val normalizedSlug = if (slug == "web-actions-v1") "web-actions" else slug ?: "null"
    val dashboardTab = dashboardTabs.firstOrNull { slug == it.slug } ?: throw NotFoundException("No tab found for slug: $normalizedSlug")
```

then dashboardTab will be smartcast as not null below.

Reply:
Fixed here https://github.com/cashapp/misk/pull/2808